src/neighbors.cc

```diff
@@ -165,14 +165,18 @@ peerAllowedToUse(const CachePeer * p, HttpRequest * request)
         return true;
 
     ACLFilledChecklist checklist(p->access, request, NULL);
-//    checklist.al = ps->al;
+    checklist.al = ps->al;
```

Comment:
If I am not mistaken, `ps` can be nil here because you did not supply `ps` in ignoreMulticastReply(). If I am right, please make this code conditional on `ps` being not nil. I would also add a comment:

```
// TODO: Make this unconditional after fixing XXX in ignoreMulticastReply().
if (ps) {
}
```

I realize we may never complete this TODO in this PR, but I think we should mark this code in case we have to come back to it later for other reasons.